src/core/caching/OperationCache.ts

```diff
@@ -10,10 +11,21 @@ export default class OperationCache {
     localStorage.setItem("operationCache", JSON.stringify(operations));
   }
 
-  static getOperations<Model>(): Operation<Model>[] {
+  static async getOperations(): Promise<Operation<SupportedModel>[]> {
     logMethodCall("OperationCache.getOperations");
     const fromCache = localStorage.getItem("operationCache");
-    const operations: Operation<Model>[] = fromCache ? Object.values(JSON.parse(fromCache)) : [];
+    const rawOperations: Operation<SupportedModel>[] = fromCache ? Object.values(JSON.parse(fromCache)) : [];
+    const operations: Operation<SupportedModel>[] = [];
+
+    for (let i = 0; i < rawOperations.length; i++) {
+      const rawOperation = rawOperations[i];
+
+      // return an operation object with correct refernces (in particular reference to the model)
+      const operation = await Operation.fromJSON(rawOperation);
```

Comment:
If this function throws, we need to handle it here. This would be fine because we only need the model reference to hydrate the model after the operation completes successfully. If the model is not found, presumably it's a different user and we don't need to hydrate it.